src/vaults/AsyncRequestManager.sol

```diff
@@ -151,7 +168,15 @@ contract AsyncRequestManager is Auth, Recoverable, IAsyncRequestManager {
     }
 
     function _sendRequest(IBaseVault vault_, bytes memory payload) internal {
-        spoke.request(vault_.poolId(), vault_.scId(), spoke.vaultDetails(vault_).assetId, payload);
+        IRefundEscrow refund = refundEscrowFactory.get(vault_.poolId());
+        require(address(refund).code.length > 0, RefundEscrowNotDeployed());
+
+        refund.withdrawFunds(address(this), address(refund).balance); // All funds goes to this contract
+
+        // It use all funds for the message, and the rest is refunded again to the RefundEscrow
+        spoke.request{
+            value: balanceSheet.gateway().isBatching() ? 0 : address(this).balance
+        }(vault_.poolId(), vault_.scId(), spoke.vaultDetails(vault_).assetId, payload, address(refund), true);
```

Comment:
I'm not fully following this. By setting `unpaid=true`, aren't we not actually paying for the tx with subsidized funds?